driver-core/src/main/com/mongodb/operation/QueryBatchCursor.java

```diff
@@ -46,7 +46,7 @@
 import static com.mongodb.operation.QueryHelper.translateCommandException;
 import static java.util.Collections.singletonList;
 
-class QueryBatchCursor<T> implements BatchCursor<T> {
+class QueryBatchCursor<T> implements ChangeStreamCursor<T> {
```

Comment:
This also feels an odd specialization.  Perhaps the cursor document as a whole could work here and be more palatable?

Reply:
I need to return the `postBatchResumeToken` from the `QueryResult` so that it can be cached in the cursor. `ChangeStreamCursor` adds a public method to return the `postBatchResumeCursor` so that it can be cached.

Reply:
I have removed `ChangeStreamCursors` and get the postBatchResumeToken from the `QueryResult`.